### PR TITLE
Fixed WorldLimiter jitter

### PR DESCRIPTION
**Pull request by LogicParrot**
Closes https://github.com/cuberite/cuberite/issues/3134

### CODE REVIEW

File: worldlimiter.lua [hunk 1]
```diff
@@ -1,29 +1,50 @@
+local WorldLimiter_ScheduleInProgress = 0
```
Comment (tigerw) [inline worldlimiter.lua]: (ಠ_ಠ)

Comment (LogicParrot) [inline worldlimiter.lua]: Thought I'd minimize the chance of collision with other globals :P

File: worldlimiter.lua [hunk 2]
```diff
@@ -1,29 +1,50 @@
...
+			NewZ = SpawnZ + WorldLimitWidth
+		elseif Z < SpawnZ - WorldLimitWidth then
+			NewZ = SpawnZ - WorldLimitWidth
 		end
+
+
+		if X ~= NewX or Z ~= NewZ then
+			WorldLimiter_ScheduleInProgress = 1
+			local X = NewX
+			local Z = NewZ
```
Comment (LogicParrot) [inline worldlimiter.lua]: I think there are some redundant leftover variables from a previous idea I had here.

File: worldlimiter.lua (hunk 1 above)
Comment (tigerw) [inline worldlimiter.lua]: (ಠ_ಠ)

Comment (LogicParrot) [inline worldlimiter.lua]: My browser cannot render the "eyes" of your textual emoji. A screenshot or a description would be appreciated.

Comment (tigerw) [inline worldlimiter.lua]: Not impressed

Comment (LogicParrot) [inline worldlimiter.lua]: :laughing:

Comment (tigerw) [inline worldlimiter.lua]: At least Lua has a `false` value right?

Comment (LogicParrot) [inline worldlimiter.lua]: Right. Forgive me, these are among my first lines of Lua. I'll fix this in a moment.

File: worldlimiter.lua [hunk 3]
```diff
@@ -1,29 +1,54 @@
...
-			Player:TeleportToCoords(Player:GetPosX(), Player:GetPosY(), Player:GetPosZ() - 1) 
+		local Limit = LimitChunks * 16 - 1
+		local SpawnX = math.floor(World:GetSpawnX())
+		local SpawnZ = math.floor(World:GetSpawnZ())
+		local X = math.floor(Player:GetPosX())
+		local Z = math.floor(Player:GetPosZ())
+		local NewX = X
+		local NewZ = Z
+		
+		if X > SpawnX + Limit then
```
Comment (NiLSPACE) [inline worldlimiter.lua]: I'd prefer to have braces around `SpawnX + Limit` just to make it more clear, but I don't know if it's required in the Core.

**Comment by NiLSPACE** [comment]
Oh, I thought you were talking about a Just In Time compiler :/

File: worldlimiter.lua (hunk 3 above)
Comment (LogicParrot) [inline worldlimiter.lua]: I added braces. Isn't a c-style `if (X > SpawnX + Limit then)` enough though?

**Comment by LogicParrot** [comment]
Squashed.

**Comment by SphinxC0re** [comment]
Looks good to me

**Comment by LogicParrot** [comment]
Merge?

**Comment by LogicParrot** [comment]
I have now write permission for this repo, by the way.

**Comment by Seadragon91** [comment]
I have no write permission, too.

**Pull request merged**